fastlane/lanes/build.rb

```diff
@@ -0,0 +1,427 @@
+    )
+  end
+
+  # Builds the Jetpack app for Testing
+  #
+  # @called_by CI
+  #
+  desc 'Build Jetpack for Testing'
+  lane :build_jetpack_for_testing do |options|
+    run_tests(
+      workspace: WORKSPACE_PATH,
+      scheme: 'Jetpack',
+      derived_data_path: DERIVED_DATA_PATH,
+      build_for_testing: true,
+      deployment_target_version: options[:ios_version]
+    )
+  end
+
+  # Runs tests without building the app.
+  #
+  # Requires a prebuilt xctestrun file and simulator destination where the tests will be run.
+  #
+  # @option [String] name The (partial) name of the `*.xctestrun` file to run
+  # @option [String] device Name of the simulator device to run the test on
+  # @option [String] ios_version The deployment target version to test on
+  #
+  # @called_by CI
+  #
+  desc 'Run tests without building'
+  lane :test_without_building do |options|
+    # Find the referenced .xctestrun file based on its name
+    build_products_path = File.join(DERIVED_DATA_PATH, 'Build', 'Products')
+
+    test_plan_path = Dir.glob(File.join(build_products_path, '*.xctestrun')).select do |e|
+      e.include?(options[:name])
+    end.first
+
+    UI.user_error!("Unable to find .xctestrun file at #{build_products_path}") if test_plan_path.nil? || !File.exist?((test_plan_path))
+
+    run_tests(
+      workspace: WORKSPACE_PATH,
+      scheme: 'WordPress',
+      device: options[:device],
+      deployment_target_version: options[:ios_version],
+      ensure_devices_found: true,
+      test_without_building: true,
+      xctestrun: test_plan_path,
+      output_directory: File.join(PROJECT_ROOT_FOLDER, 'build', 'results'),
+      reset_simulator: true,
+      result_bundle: true
+    )
+  end
+
+  # Builds the app and uploads it to TestFlight / App Store Connect
+  #
+  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
+  # @option [Boolean] create_gh_release If tru, creates a GitHub Release draft after the upload, with zipped xcarchive as artefact
+  # @option [Boolean] beta_release If true, the GitHub release will be marked as being a pre-release
+  #
+  # @called_by CI
+  # @calls build_and_upload_itc
+  #
+  desc 'Builds and uploads for distribution to App Store Connect'
+  lane :build_and_upload_app_store_connect do |options|
+    ios_build_prechecks(
+      skip_confirm: options[:skip_confirm],
+      internal: options[:beta_release],
+      external: true
+    )
+
+    ios_build_preflight
+
+    build_and_upload_itc(
+      skip_prechecks: true,
+      skip_confirm: options[:skip_confirm],
+      beta_release: options[:beta_release],
+      create_release: options[:create_gh_release]
+    )
+  end
+
+  # Builds the Jetpack app and uploads it to TestFlight, for beta-testing or final release
+  #
+  # @called_by CI
+  #
+  desc 'Builds and uploads Jetpack to TestFlight for distribution'
+  lane :build_and_upload_jetpack_for_app_store do
+    jetpack_appstore_code_signing
```

Comment:
I think it would be useful to run the Sentry CLI check here, too, like `build_and_upload_installable_build ` does:

```suggestion
    sentry_check_cli_installed

    jetpack_appstore_code_signing
```